test_plans/memcpy2d.asciidoc

```diff
@@ -0,0 +1,165 @@
+:sectnums:
+:xrefstyle: short
+
+= Test plan for 2D memcpy and associated routines that can copy a specified rectangular region in the presence of array padding.
+
+This is a test plan for the APIs described in
+https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/supported/sycl_ext_oneapi_memcpy2d.asciidoc[SYCL_EXT_ONEAPI_memcpy2d.asciidoc]
+
+
+== Testing scope
+
+=== Device coverage
+
+All of the tests described below are performed only on the default device that
+is selected on the CTS command line.
+
+=== Feature test macro
+
+All of the tests should use `#ifdef SYCL_EXT_ONEAPI_MEMCPY2D` so they can be skipped
+if feature is not supported.
+
+=== Type coverage
+
+All the tests described below are executed for the following types unless otherwise specified:
+
+* `char`
+* `short`
+* `int`
+* `long long`
+* `std::size_t`
+* `bool`
+* `float`
+* A user-defined trivially-copyable struct with multiple member variables.
+
+In addition, if the device has aspect `aspect::fp64`:
+
+* `double`
+
+In addition, if the device has `aspect::fp16`:
+
+* `sycl::half`
+
+== Tests
+
+=== Handler class methods
+
+==== ext_oneapi_memcpy2d
+
+`void ext_oneapi_memcpy2d(void *dest, size_t destPitch,
+  const void *src, size_t srcPitch,
+  size_t width, size_t height)`
+
+Check function with combinations of `host pointer` and `pointer within device USM allocation` types of `src` and `dest`.
+In case of pointer within device USM allocation and the selected device lacks this capability
+(aspect `aspect::usm_device_allocations`), this test is skipped.
+
+Type of `src` and `dest` buffers elements is `unsigned char`.
+
+Checks:
+
+* Set value of `width` large than `srcPitch` and check if function throws a synchronous exception with the `errc::invalid` error code.
+* Set value of `width` large than `destPitch` and check if function throws a synchronous exception with the `errc::invalid` error code.
+* `src` buffer elements should be initialized with `expected_val`. `dest` buffer elements should be initialized with `init_val`.
+`width` should satisfy the conditions: `width` < `destPitch` and `width` <= `srcPitch`. Check after performing `ext_oneapi_memcpy2d`
```

Comment:
```suggestion
`width` should satisfy the conditions: `width` < `destPitch` and `width` +<=+ `srcPitch`. Check after performing `ext_oneapi_memcpy2d`
```

Without this, asciidoctor renders this as a left-arrow in HTML.

Reply:
Done. Thanks!